### PR TITLE
Refactor for --type support of ee site command

### PR DESCRIPTION
**Pull request by mrrobot47**
Test: https://travis-ci.org/mrrobot47/easyengine/builds/421700140

### CODE REVIEW

File: AcmePhp/Cli/Exception/AcmeCliActionException.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<?php
```
Comment (mrrobot47) [inline AcmePhp/Cli/Exception/AcmeCliActionException.php]: Moved from EE core to here.

File: AcmePhp/Cli/Exception/AcmeCliException.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<?php
```
Comment (mrrobot47) [inline AcmePhp/Cli/Exception/AcmeCliException.php]: Moved from EE core to here.

File: AcmePhp/Cli/Exception/AcmeDnsResolutionException.php [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<?php
```
Comment (mrrobot47) [inline AcmePhp/Cli/Exception/AcmeDnsResolutionException.php]: Moved from EE core to here.

File: AcmePhp/Cli/Exception/CommandFlowException.php [hunk 4]
```diff
@@ -0,0 +1,55 @@
+<?php
```
Comment (mrrobot47) [inline AcmePhp/Cli/Exception/CommandFlowException.php]: Moved from EE core to here.

File: AcmePhp/Cli/Repository/Repository.php [hunk 5]
```diff
@@ -0,0 +1,439 @@
+<?php
```
Comment (mrrobot47) [inline AcmePhp/Cli/Repository/Repository.php]: Moved from EE core to here.

File: AcmePhp/Cli/Repository/RepositoryInterface.php [hunk 6]
```diff
@@ -0,0 +1,198 @@
+<?php
```
Comment (mrrobot47) [inline AcmePhp/Cli/Repository/RepositoryInterface.php]: Moved from EE core to here.

File: AcmePhp/Cli/Repository/RepositoryV2Interface.php [hunk 7]
```diff
@@ -0,0 +1,51 @@
+<?php
```
Comment (mrrobot47) [inline AcmePhp/Cli/Repository/RepositoryV2Interface.php]: Moved from EE core to here.

File: AcmePhp/Cli/Serializer/PemEncoder.php [hunk 8]
```diff
@@ -0,0 +1,55 @@
+<?php
```
Comment (mrrobot47) [inline AcmePhp/Cli/Serializer/PemEncoder.php]: Moved from EE core to here.

File: AcmePhp/Cli/Serializer/PemNormalizer.php [hunk 9]
```diff
@@ -0,0 +1,55 @@
+<?php
```
Comment (mrrobot47) [inline AcmePhp/Cli/Serializer/PemNormalizer.php]: Moved from EE core to here.

File: src/helper/Shutdown_Handler.php [hunk 10]
```diff
@@ -0,0 +1,21 @@
+<?php
```
Comment (mrrobot47) [inline src/helper/Shutdown_Handler.php]: Moved from EE core to here.

File: src/helper/Site_Letsencrypt.php [hunk 11]
```diff
@@ -0,0 +1,529 @@
+<?php
```
Comment (mrrobot47) [inline src/helper/Site_Letsencrypt.php]: Moved from EE core to here. But has namespace related changes.

File: src/helper/class-ee-site.php [hunk 12]
```diff
@@ -0,0 +1,543 @@
+<?php
```
Comment (mrrobot47) [inline src/helper/class-ee-site.php]: Moved from EE core to here. But has namespace related changes.

File: src/helper/site-utils.php [hunk 13]
```diff
@@ -0,0 +1,371 @@
+<?php
```
Comment (mrrobot47) [inline src/helper/site-utils.php]: Moved from EE core to here. But has namespace related changes.

File: templates/global_docker_compose.yml.mustache [hunk 14]
```diff
@@ -0,0 +1,54 @@
+version: '3.5'
```
Comment (mrrobot47) [inline templates/global_docker_compose.yml.mustache]: Moved from EE core to here.

File: site-command.php [hunk 15]
```diff
@@ -13,4 +13,11 @@
 	require_once $autoload;
 }
 
+// Load utility functions.
+require_once 'src/helper/site-utils.php';
```
Comment (rahulsprajapati) [inline site-command.php]: add file using abs path.
Use __DIR__ . '/src/helper/site-utills.php';
Do same for other places also.

**Pull request merged**